client/src/components/molecules/UnmatchedDonationFormsTable.tsx

```diff
@@ -256,16 +256,18 @@ const UnmatchedDonationFormsTable: FunctionComponent<Props> = (props: Props) =>
                         <td />
                         <td />
                     </tr>
-                    {donationForms.map((donationForm: DonationForm) => (
+                    {donationForms.map((donationForm: DonationForm, index) => (
                         <>
                             <tr
                                 key={donationForm._id}
                                 className="data-row"
                                 onClick={() => {
                                     setSelectedDonationFormForInspection(donationForm);
+                                    donationForm.seen = true;
```

Comment:
Instead of directly setting `donationForm.seen = true`, you'll need to use `setDonationForms` 

I recommend writing a function like this
```
const setDonationFormSeen = (donationFormId) => {
  const newDonationForms = donationForms.map( donationForm => donationFrom.id === donationFormId ? {...donationForm, seen: true}: donationFrom )
 
  setDonationForms(newDonationForms)
}

```